### DISCUSSION TITLE
alternative installers

### DISCUSSION THREAD

**Discussion by rdmueller** [docToolchain/docToolchain#540] [Ideas]
It might be nice to install docToochain not only via the wrapper but also via homebrew and other tools.

**Comment by rdmueller** [comment]
docToolchain can already be installed via sdkman.io

**Reply by joel-loewenstein** [reply]
why does https://sdkman.io/sdks#doctoolchain say coming soon?
i was just about to clean up my system and do it the sdkman way, just did not find a fast way how to understand what it is acually perfoming when initiating `$ sdk install doctoolchain` just when you mentioned below you transport a zip file...

**Reply by rdmueller** [reply]
sdkman installs everything in the home folder - that is quite nice.
The dtc wrapper does the same.
sdk says "coming soon" because I haven't fully tested it yet and thus it only has beta versions available.
but I am currently doing manual zests for the different installers

**Comment by joel-loewenstein** [comment]
I was thinking of only installing the NG wrapper via a brew formulae. Otherwise it would become complex to follow their strategy of reusing fomulae like ascidoctorj and gradle, which will or might depending on setting automatically update to newer versions.

**Reply by rdmueller** [reply]
sounds like a plan. For sdkman, we distribute docToolchain as CLI in a zip file

**Reply by joel-loewenstein** [reply]
I would suggest that the wrapper defines one installation method for CLI and Docker-based, and can be triggered from brew, sdk and even other package managers who would usually update the wrapper only. 

Which in turn based on the github repo would take all necessary steps to install what is needed for the actual version: 
- java
- gradlew
- asciidoc
- ascidoc-pdf
- etc.. 

Otherwise dtc might end up with serveral detailed installation packages depending on the structure of the package manager. 

All those tools should acutally be installed where they usually belong on the system instead of the home folder of the user. 
Also all paths shall be set correctly to call it. 

Then only the project related config and source should exist within that folder... 
Even then it becomes quite a complex flow i admit.

I created a draft use case diagram and block diagram in PlantUML because i need to practice and created another issue in regards to the doc being extended architecture wise, see #545.

Use case diagram | Block diagram
------------------|----------------
![Use Case Diagram](https://kroki.io/plantuml/svg/eNqNU0FOwzAQvPsVq95TKQ6nCEGrFmhFkZDaigsXk6yoVceObJe0iMdjJw5Nkx64RTOzuzO7zsRYpu2hEMTsuSyZZgUUSqpsp1WBYPUBO8yOybzS3FqUfYplVum1PQkEVqFxxYSkW4M6BWZgGpN04bAPjRWcUUrS9fy5YLKDJYQcDGbMIGxn8Tj24GgpnU8hSBSRSrOyRA28gQzkKtsoJbId4xLeJQiVMQHKKzy3R01Glz1ptyd8cQbOhe-dYynUCdoRSopTvzYZ1La5_tWAhkAubw9vTXHLmeDfA_pvE63Az3s4cmO5_IQxc1HdqMsSeqVkqrMbChssSsEs9t018Z5QonZsnwweWtq3e50_wqBJGNzVLTYvqyvCZCB0OY61sLl-dBeeQQqz1bLeKGisX229_bdm1fdBfvsTReHGKeDRPdTc9KmkQ5HmJIFzG-sDSSOhHck4HkK0kSWdwiC7gOgQcgOmYd7UmzYlZs21zJlIeoRn2s0QVxa-E-KU4ZueRd4xmbi87j__BYCAJJA=) | ![Block Diagram](https://kroki.io/plantuml/svg/eNp1UEFugzAQvPOKVU7lEBSiVj0lUnroB8oHVmZDXOy1tTZFatW_lwRCDVVv45nxeMah1exR0IJ17NRFnCWI0lEWZkWQ20Ae9ruErL2G8illUMT11UWrlikE2BXPiehRtdhQpaOhk9ENW-IIhs4xMXWBFAZ6cVKTpEmP6cN0xs7EV8fxTX8SlPu0KqmI3Jg_GWWWzRpsmHroBb0n2cBXBvBQR9Xn2ffCZVHzXXaqcs6oy0DlV6IRrA31NxwxtCHPVreVs97xsDJMGRiU1kNQdJKvzltfn5PYG_TIg7juJB1HbWmKfMcPvDnG_rDdHtddl8fRMZefUcoP9LgIiqJI2QkM3qt5Med_bZy20u_bpjVwOBzh979-ALd-2vE=)

**Comment by joel-loewenstein** [comment]
what do you think about dtcw.sh or dtc.sh domain as a starting point? like brew.sh ... 
it is still available, although .sh is not the cheapest. 
i would donate the first year... 22€ (domain only... )

**Reply by rdmueller** [reply]
nice domain. we also habe some community money available...

**Reply by joel-loewenstein** [reply]
so shall we do something about it? as said i would donate the first year. the minimum i could do to appreciate all the work you did invest in this.

**Reply by rdmueller** [reply]
where and how can the domain be registered? I normally use df.eu, but it seems that they don't register .sh domains.

we currently have a bit over 400$ in our budget: https://opencollective.com/doctoolchain
we could route this expense through opencollective.

**Reply by joel-loewenstein** [reply]
i usually use TLD-List to find out:
https://tld-list.com/tld/sh
for .sh the cheapest one 22€ is porkbun, somewhat recommended but US based with all the typical data privacy and legal hurdles... i could just buy it with my account there and then we figure out the rest later. i promise i will give you authority once i have figured out how to do that in porkbun. My initial idea would be a redirector dtc.sh -> dtc official docs site and dtc.sh/wrapper to the wrapper dl you have. or do you prefer dtcw.sh? what do you think?

Porkbun supports: DNS, Email Forwarding, SSL Certificate, WHOIS Privacy

**Reply by joel-loewenstein** [reply]
> https://opencollective.com/doctoolchain

So how would this be done through opencollective? I am totally new to it?

**Reply by rdmueller** [reply]
you can sponsor this project through opencollective - this way, I could order the domain and you pay for the first year :-)
but you can also submit expenses and take out money for the domain renewal. This way, you could register the domain and submit expenses in the following years.

**Reply by joel-loewenstein** [reply]
so what is your preference? in the end dtc or dtcw is your baby, you should decide what you think fits best.

**Reply by rdmueller** [reply]
it belongs to the community, but I think `dtcw.sh` would fit best, wouldn't it? It it hosts the wrapper.
If we want to put up a full webpage with the docs, `dtc.sh` would be a better fit. What would be the download URL then? `dtc.sh/wrapper`?

**Comment by joel-loewenstein** [comment]
> It might be nice to install docToochain not only via the wrapper but also via homebrew and other tools.

I think #278 should be mentioned here, as it contains a lot of thoughts around the topic already and i created #544 to improve dtcw with an interactive mode
